front/lib/resources/vault_resource.ts

```diff
@@ -56,23 +77,42 @@ export class VaultResource extends BaseResource<VaultModel> {
           workspaceId: workspace.id,
         },
       })
-    ).map((vault) => new this(VaultModel, vault.get()));
+    ).map(
+      (vault) =>
+        new this(
+          VaultModel,
+          vault.get(),
+          vault.groups.map(
+            (group) => new GroupResource(GroupModel, group.get())
+          )
+        )
+    );
     const systemVault =
       existingVaults.find((v) => v.kind === "system") ||
-      (await VaultResource.makeNew({
-        name: "System",
-        kind: "system",
-        workspaceId: workspace.id,
-        groupId: systemGroup.id,
-      }));
+      (await VaultResource.makeNew(
+        {
+          name: "System",
+          kind: "system",
+          workspaceId: workspace.id,
+        },
+        systemGroup
+      ));
+
     const globalVault =
       existingVaults.find((v) => v.kind === "global") ||
-      (await VaultResource.makeNew({
-        name: "Workspace",
-        kind: "global",
-        workspaceId: workspace.id,
-        groupId: globalGroup.id,
-      }));
+      (await VaultResource.makeNew(
+        {
+          name: "Workspace",
+          kind: "global",
+          workspaceId: workspace.id,
+        },
+        globalGroup
+      ));
+    await GroupVaultModel.findOrCreate({
```

Comment:
If I read the code correctly, this is already done in the `VaultResource.makeNew()` static method, right?